### PR TITLE
Record metrics when framework returns unschedulable error

### PR DESCRIPTION
**Pull request by mrkm4ntr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Unschedulable errors by scheduling framework should be recorded as metrics.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -560,6 +560,7 @@ func (sched *Scheduler) scheduleOne() {
 		if !permitStatus.IsSuccess() {
 			var reason string
 			if permitStatus.Code() == framework.Unschedulable {
```
Comment (draveness) [inline pkg/scheduler/scheduler.go]: NOTE:

> 	// Unschedulable is used when a plugin finds a pod unschedulable.
	// The accompanying status message should explain why the pod is unschedulable.
	Unschedulable

**Review by draveness — APPROVED** [review]
/ok-to-test
/priority important-soon
/lgtm

**Comment by mrkm4ntr** [comment]
@Huang-Wei Could you review this?

**Review by wgliang — APPROVED** [review]

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -560,6 +560,7 @@ func (sched *Scheduler) scheduleOne() {
...
+				metrics.PodScheduleFailures.Inc()
```
Comment (Huang-Wei) [inline pkg/scheduler/scheduler.go]: This doesn't look correct. For each scheduling cycle, we should at most log one failure - which is conditionally performed at:

https://github.com/kubernetes/kubernetes/blob/19e85a9092399b065309aa5b4aec71a5b8101bab/pkg/scheduler/scheduler.go#L504

We don't and shouldn't log another failure here.

Comment (mrkm4ntr) [reply pkg/scheduler/scheduler.go]: Thank you for your review.

> For each scheduling cycle, we should at most log one failure - which is conditionally performed at: https://github.com/kubernetes/kubernetes/blob/19e85a9092399b065309aa5b4aec71a5b8101bab/pkg/scheduler/scheduler.go#L504

But that path immediately returns, so each scheduling cycle logs at most once. According to the definition of metrics, I think we should increment `scheduleAttempts` with any label in one scheduling cycle.
https://github.com/kubernetes/kubernetes/blob/22cf3ca0a536b402a00ca771960702c811a3ca5c/pkg/scheduler/metrics/metrics.go#L58-L63

In this case, I think `unschedulable` is appropriate.

Comment (Huang-Wei) [reply pkg/scheduler/scheduler.go]: Hmm... I'm a little surprised we allow/expect Permit plugin to return "Unschedulable" b/c Permit/Bind plugins belong to "Binding Cycle" which shouldn't be considered as a "schedule failure", in my opinion.

But based on current codebase, your PR is correct, we should increase PodScheduleFailures metric by 1 in the `if` block. Thanks.

I will follow up with @ahg-g to see if we should eliminate the logic related with tolerating/expecting Premit/Bind plugins to return "unschedulable" status.

Comment (ahg-g) [reply pkg/scheduler/scheduler.go]: Yeah, Permit unfortunately is different. When a pod is rejected (by other pod, or timeout) this is not considered a scheduling Error, and so we used unschedulable. Do you have other suggestions we might be able to handle this case?

**Pull request merged**